zebra-chain/src/block.rs

```diff
@@ -69,24 +69,23 @@ impl Block {
     ///
     /// "The first (and only the first) transaction in a block is a coinbase
     /// transaction, which collects and spends any miner subsidy and transaction
-    /// fees paid by transactions included in this block."[S 3.10][3.10]
+    /// fees paid by transactions included in this block." [§3.10][3.10]
     ///
     /// [3.10]: https://zips.z.cash/protocol/protocol.pdf#coinbasetransactions
     pub fn is_coinbase_first(&self) -> Result<(), Error> {
-        if self.coinbase_height().is_some() {
-            // No coinbase inputs in additional transactions allowed
-            if self
-                .transactions
-                .iter()
-                .skip(1)
-                .any(|tx| tx.contains_coinbase_input())
-            {
-                Err("coinbase input found in additional transaction")?
-            }
-            Ok(())
-        } else {
-            Err("no coinbase transaction in block")?
+        let first = self
+            .transactions
+            .iter()
+            .next()
+            .ok_or_else(|| "block has no transactions")?;
+        let mut rest = self.transactions.iter().skip(1);
```

Comment:
```suggestion
        let mut trans = self.transactions.iter();
        let first = trans
            .next()
            .ok_or_else(|| "block has no transactions")?;
        let mut rest = trans;
```

Reply:
Changed to `self.transactions.get(0)` to satisfy clippy; I think it's better not to have a binding for the iterator because I don't like to bind variables unless I can give them meaningful names.